app/src/main/java/me/quadphase/qpdex/AdvancedSearch.java

```diff
@@ -0,0 +1,508 @@
+        }else {
+            selectedSpdefGreater = max_stat;
+        }
+
+        if (!speedLowerTextUI.getText().toString().equals("")){
+            selectedSpeedLower = Integer.parseInt(speedLowerTextUI.getText().toString());
+        }else{
+            selectedSpeedLower = min_stat;
+        }
+
+        if (!speedGreaterTextUI.getText().toString().equals("")){
+            selectedSpeedGreater = Integer.parseInt(speedGreaterTextUI.getText().toString());
+        }else {
+            selectedSpeedGreater = max_stat;
+        }
+
+        Log.d("QPDEX", String.format("Stats chosen: %s/%s/%s/%s/%s/%s - %s/%s/%s/%s/%s/%s ",
+                String.valueOf(selectedHpLower),String.valueOf(selectedAttLower),String.valueOf(selectedDefLower),
+                String.valueOf(selectedSpattLower),String.valueOf(selectedSpdefLower),String.valueOf(selectedSpeedLower),
+                String.valueOf(selectedHpGreater),String.valueOf(selectedAttGreater),String.valueOf(selectedDefGreater),
+                String.valueOf(selectedSpattGreater),String.valueOf(selectedSpdefGreater),String.valueOf(selectedSpeedGreater)
+                ));
+
+    }
+
+    /**
+     * Function to filter for the stats of pokemon
+     *
+     * @param lowerLimit of the stat
+     * @param upperLimit of the stat
+     * @param statName either HP, DEFENCE, ATTACK, SPATTACK, SPDEFENCE or SPEED
+     */
+    private void filterStats(int lowerLimit, int upperLimit, String statName){
+
+        // Filter only if one of the values is not a default value
+        if(lowerLimit > 0 || upperLimit < 999){
+            ArrayList<Integer> tempFilterPokemonArrayList = new ArrayList<Integer>();
+            tempFilterPokemonArrayList = pokemonFactory.getAllUniqueIDsFromStat(lowerLimit, upperLimit, statName);
+            currentFilterPokemonArrayList.retainAll(tempFilterPokemonArrayList);
+            tempFilterPokemonArrayList.clear();
+        }
+
+    }
+
+
+
+
+
+
+    public void switchToPokedex(View view){
+        long startTime = System.nanoTime();
+        Intent intent = new Intent(this,PokedexActivity.class);
+
+        // Obtain all stats chosen
+        collectStats();
+
+        //Comment this out, takes way tooo long 4+sec
+        currentFilterPokemonArrayList.clear();
+        currentFilterPokemonArrayList.addAll(filteredUniqueIds);
+
+        ArrayList<Integer> tempFilterPokemonArrayList = new ArrayList<>();
+
+        // Filtering algorithm, will go through a list of all uniqueIDs and apply filters
+        // consecutively. Once the list is filtered, convert the unique to nationalIds
+
+        //Start with the ability chosen, if any. Use the ability ID to get all NationalID's associated
+        if(selectedAbility.getName() != "None"){
+
+            currentFilterPokemonArrayList = pokemonFactory.getAllUniqueIDsFromAbility(Arrays.asList(allAbilities).indexOf(selectedAbility));

```

Comment:
I agree with this refactoring. It should definitely be done, it makes life a lot easier.